push.md

```diff
@@ -1,27 +1,201 @@
-# Push Service
+# Push Service (WebSockets)
 
-The GroupMe Push Service is strange and finnicky, so giving precise documentation about it is hard.
-The [official documentation](https://dev.groupme.com/tutorials/push) has a lot of outdated and incomplete information,
-and it's hard (for me, at least) to figure out what's required and what's not.
-
-Hopefully someone can figure out how it works and add some details here, but until then, here are some working
-implementations:
-
-* [My bot, Lowes.](https://github.com/2CATteam/gmuserbot/blob/master/bot.js) I'm sure I stole this working code from
-  somewhere, but I can't for the life of me figure out where.
-
-* [groupme_push](https://github.com/cuuush/groupme-push), a groupme push implementation in Python. 
-
-* [The GroupMe npm package](https://github.com/njoubert/node-groupme/blob/master/lib/IncomingStream.js)
+The GroupMe Push Service is strange and finicky, so giving precise documentation about it is hard. 
+It's built off of [Faye](https://faye.jcoglan.com/browser/subscribing.html), a relatively simple pub/sub websocket messaging protocol. 
+You may want to read its docs for a deeper understanding of how it works.
+The [official documentation](https://dev.groupme.com/tutorials/push) has some outdated and incomplete information,
+it may be hard to figure out what's required and what's not.
 
 Here is a list of some known WebSocket types and what they correspond to:
 
 * `ping` -> A ping from the web socket server
 * `line.create` -> New Group Message. This includes anything that makes a message such as calendar events,
   joining/leaving events, polls, etc.
 * `like.create` -> Someone likes a message that is yours in a group
```

Comment:
How do `like.create` and `favorite` work in the era of reactions? Are these still called `like` and `favorite` despite containing reaction data?

Reply:
Yep, they behave exactly the same except now they contain a `reaction` parameter to define which reaction was applied.

Reply:
Awesome, good to know. In that case I think this is all set to merge 👍